src/test/java/oop/animals/enclosures/EnclosureTest.java

```diff
@@ -0,0 +1,116 @@
+package oop.animals.enclosures;
+
+import oop.animals.model.mammals.Mammal;
+import oop.animals.model.mammals.predators.Predator;
+import oop.animals.model.mammals.predators.aquatic.Cachalot;
+import oop.animals.model.mammals.predators.aquatic.KillerWhale;
+import oop.animals.model.mammals.predators.ground.Coyote;
+import oop.animals.model.mammals.predators.ground.Lion;
+import oop.animals.model.mammals.predators.mixed.JungleCat;
+import oop.animals.model.mammals.predators.mixed.WhiteBear;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class EnclosureTest {
+    private Lion lion1, lion2;
+    private Coyote coyote;
+    private Cachalot cachalot;
+    private KillerWhale killerWhale;
+    private JungleCat jungleCat;
+    private WhiteBear whiteBear;
+    private Enclosure<Mammal> enclosure;
+
+    @BeforeEach
```

Comment:
I think we don't need to recreate animals before each test, since they are not changing in tests. Please add constructors with parameters and empty constructors as well for all animals and create test objects in place where you declare variables.